### PR TITLE
GB-1288 

### PR DESCRIPTION
**Pull request by lonellliburd**

### CODE REVIEW

File: flux/src/main/webapp/jsp/includes/navLink.jsp [hunk 1]
```diff
@@ -40,12 +41,16 @@
...
+		$.post("${pageContext.request.contextPath}/moreRefinements.html", {
+			"navigationName" : navigationName,
+			"selectedRefinements": getSelectedRefinements()
+		}).done(function(data){
 			if (data != "" || data != undefined) {
-				$(".nav-" + navigationName).replaceWith(data);
+				$('#nav-'+navigationName).replaceWith(data);
 				$("#more-" + navigationName).fadeOut();
 			} else {
 				location.reload();
```
Comment (willwarren) [inline flux/src/main/webapp/jsp/includes/navLink.jsp]: replace with console.log

File: flux/src/main/webapp/js/main.js [hunk 2]
```diff
@@ -38,4 +38,24 @@ function prettyJson() {
...
+    var selected = $('#refinements').val().split("~");
+    var unrefined = selected.sanitize(selected);
+    var refined = [];
+    for(var i=0; i<unrefined.length; i++){
+        refined.push(unrefined[i].split("=")[1]);
+    }
+    return refined.toString();
+}
+
+Array.prototype.sanitize = function() {
```
Comment (willwarren) [inline flux/src/main/webapp/js/main.js]: maybe extract to a function.

File: flux/src/main/java/com/groupbyinc/quickstart/controller/NavigationController.java [hunk 3]
```diff
@@ -45,46 +44,45 @@
...
 
+        Results results = new Results();
+        results.setSelectedNavigation(Utils.getSelectedNavigations(selectedRefinements));
         RefinementsResult refinementsResults = bridge.refinements(
-                new Query().setArea(query.getArea())
-                           .addRefinementsByString(query.getRefinementString()) //
-                           .setCollection(query.getCollection()) //
-                           .addValueRefinement("gbi_stream_upload", "1"), navigationName); //
+                new Query().setArea(query.getArea()).addRefinementsByString(query.getRefinementString()).setCollection(
+                        query.getCollection()).addValueRefinement("gbi_stream_upload", "1"), navigationName);
```
Comment (willwarren) [inline flux/src/main/java/com/groupbyinc/quickstart/controller/NavigationController.java]: Why are we addiding a value refinement here?

**Comment by willwarren** [comment]
@lonellliburd quick change

File: flux/src/main/java/com/groupbyinc/quickstart/controller/NavigationController.java (hunk 3 above)
Comment (lonellliburd) [inline flux/src/main/java/com/groupbyinc/quickstart/controller/NavigationController.java]: I am not entirely sure; it was there before and when I took it out, the whole thing crashed.

Comment (willwarren) [inline flux/src/main/java/com/groupbyinc/quickstart/controller/NavigationController.java]: @effervescentia do you know?

Comment (effervescentia) [inline flux/src/main/java/com/groupbyinc/quickstart/controller/NavigationController.java]: @willwarren @lonellliburd This field is no longer placed on the records (once upon a time it was to compensate for the GSA being unable to do a search with a blank query). There are, oddly enough, still bits of code in bindle that reference this field though (which is worrisome). Removing that one refinement shouldn't break everything.

Comment (lonellliburd) [inline flux/src/main/java/com/groupbyinc/quickstart/controller/NavigationController.java]: @effervescentia it breaks the refinement retrieval portion of the code. I took it out and it it wouldn't work otherwise.

Comment (effervescentia) [inline flux/src/main/java/com/groupbyinc/quickstart/controller/NavigationController.java]: @lonellliburd What's the actual error?

File: flux/src/main/java/com/groupbyinc/quickstart/controller/NavigationController.java [hunk 4]
```diff
@@ -45,46 +44,45 @@
...
+            return new ModelAndView("includes/navLink.jsp", model);
         }
 
+        Results results = new Results();
+        results.setSelectedNavigation(Utils.getSelectedNavigations(selectedRefinements));
         RefinementsResult refinementsResults = bridge.refinements(
-                new Query().setArea(query.getArea())
-                           .addRefinementsByString(query.getRefinementString()) //
-                           .setCollection(query.getCollection()) //
-                           .addValueRefinement("gbi_stream_upload", "1"), navigationName); //
```
Comment (lonellliburd) [inline flux/src/main/java/com/groupbyinc/quickstart/controller/NavigationController.java]: @effervescentia So I removed it and ran over the tests for edge cases Will had me do. The error doesn't show up anymore. It might have been a combination of removing this and another bug that caused the error so once I resolved the other bugs this one resolved as well

Comment (effervescentia) [inline flux/src/main/java/com/groupbyinc/quickstart/controller/NavigationController.java]: Excellent :dancers:
